### ISSUE TITLE
[Bug] Cannot read property 'message' of null

### ISSUE THREAD

**Issue by Passpes** [grafana/grafana#4499]
Hi folks,

I'm new to Grafana so excuse me if i don't use the right words to describe my issue but i'm trying to install Grafana+Graphite+Icinga2 solution so I'm getting all the metrics i need in Graphite browser from Icinga2, the problem is when i created a new data source to get this metrcis to Grafana this problem occure, searched for this issue and found that is related to node js smthing like this.
Need some help on this people! show me your talents.
![garafana1](https://cloud.githubusercontent.com/assets/6490137/14110155/989bca72-f5c5-11e5-8c5d-4de8393a0fb0.PNG)
![grafana2](https://cloud.githubusercontent.com/assets/6490137/14110160/9e05231e-f5c5-11e5-8b0f-4160285c9cd0.PNG)

**Comment by torkelo** [comment]
try Grafana 3.0

**Comment by torkelo** [comment]
this this if fixed

**Comment by Satisheupure** [comment]
I am getting same error for elasticsearch

**Comment by torkelo** [comment]
It's probably a CORS error as you are using direct access, have you enabled CORS in elasticsearch.yml config?

**Comment by Satisheupure** [comment]
I already have below in elasticsearch.yml . 

http.cors.enabled: true
http.cors.allow-origin: "*"

I followed steps from grafana site to setup elasticsearch.
Added elastic search source and tested successfully.

In query editor as soon as I click on the +Query in the top left cornor I get this error "Cannot read property 'message' of null" I can see fields from elastic search. But what ever I do this error is not going away.

I definitely have data in elasticsearch which I verified with other plugin's

Please help I don't want to switch to kibana.

**Comment by torkelo** [comment]
Can you show me you elasticsearch query and you es document, also please show elactic request and response (chrome dev tools network tab)

**Comment by Satisheupure** [comment]
![es_datasource](https://cloud.githubusercontent.com/assets/18171945/14406382/06485f98-fe74-11e5-92a0-8711b4fd682d.jpg)

As soon as I add the datasource and click on +Query I am getting the error
![es_query](https://cloud.githubusercontent.com/assets/18171945/14406425/1149e6c2-fe75-11e5-942f-cfbbb475f3b3.jpg)

Logstash conf:

input {
  file {
    path => [ "C:\Jmeter_test_cases\test_results.csv"]
  }
}
filter {
  if ([message] =~ "responseCode") {
    drop { }
  } else {
    csv {
    	columns => ["time","elapsed", "label", "success", "bytes", "grpThreads", "allThreads", "Latency",
	            "SampleCount", "ErrorCount", "Hostname"]
    }
  }
}
output {
  	elasticsearch {
  	hosts => "localhost:9200"
    	index => "jmeter-results-%{+YYYY.MM.dd}"
    	template => "C:\Jmeter_test_cases\jmeter-results-mapping.json"
     	template_name => "logstash-jmeter-results"
 	template_overwrite => true
  }
}

Below is the template from above script(template => "C:\Jmeter_test_cases\jmeter-results-mapping.json"):


{
    "template": "jmeter-results-*",
    "settings": {
        "number_of_shards": 1,
        "number_of_replicas": 0,
        "index.refresh_interval": "5s"
    },
    "mappings": {
        "logs": {
            "properties": {
                "time": {
                     "type": "date",
                     "index" : "not_analyzed",
		     "format" : "yyyy-MM-dd HH:mm:ss.SSS" 
                },
                "elapsed": {
                    "type": "long"
                },
                "label": {
                    "type": "string"
                },
                "success": {
                    "type": "boolean"
                },
                "bytes": {
                    "type": "long"
                },
                "grpThreads": {
                    "type": "long"
                },
                "allThreads": {
                    "type": "long"
                },
                "Latency": {
                    "type": "long"
                },
                "SampleCount": {
                    "type": "long"
                },
                "ErrorCount": {
                    "type": "long"
                },
                "Hostname": {
                    "type": "string"
                }
            }
        }
    }
}

ES_request with just ES source added

![es_request1](https://cloud.githubusercontent.com/assets/18171945/14406515/f8c4ee96-fe77-11e5-9b19-30de3a1030ea.jpg)

ES_request after clicking on +Query no changes to query and I am getting the error.
The request and response tabs are empty on Chome dev tools.

![es_request2](https://cloud.githubusercontent.com/assets/18171945/14406521/17be9bc6-fe78-11e5-85af-73c2705937ca.jpg)

**Comment by Satisheupure** [comment]
Made some changes to query
![es_request3](https://cloud.githubusercontent.com/assets/18171945/14406567/8b567062-fe79-11e5-8720-f125937e7f89.jpg)

Chome Dev stools preview tab:

![es_request4](https://cloud.githubusercontent.com/assets/18171945/14406570/a5bb38f2-fe79-11e5-8f98-5b02a599ebf2.jpg)

Chome Dev tools Response tab code:

{"jmeter-results-2016.04.09":{"mappings":{"logs":{"properties":{"@timestamp":{"type":"date","format":"strict_date_optional_time||epoch_millis"},"@version":{"type":"string"},"ErrorCount":{"type":"long"},"Hostname":{"type":"string"},"Latency":{"type":"long"},"SampleCount":{"type":"long"},"allThreads":{"type":"long"},"bytes":{"type":"long"},"elapsed":{"type":"long"},"grpThreads":{"type":"long"},"host":{"type":"string"},"label":{"type":"string"},"message":{"type":"string"},"path":{"type":"string"},"success":{"type":"boolean"},"time":{"type":"date","format":"yyyy-MM-dd HH:mm:ss.SSS"}}}}}}

**Comment by torkelo** [comment]
The json response from the msearch request would help

**Comment by Satisheupure** [comment]
Is this what you need?

![es_msearch](https://cloud.githubusercontent.com/assets/18171945/14406750/8ea72242-fe7f-11e5-9855-5c324fdbf507.jpg)

**Comment by torkelo** [comment]
Yea, never seen that elasticsearch error, you will have to search  elasticsearch support forums / Google for that one

**Comment by ke4roh** [comment]
I am having this error with a Graphite source on latest, 3.0.0-beta5, and head (which is only a few hours old).   It is pretty easy to reproduce with [this docker environment](https://github.com/jmreicha/graphite-docker). Just edit the docker-compose.yml to indicate which version of Graphite to try.

**Comment by torkelo** [comment]
When do you get this error ? What query ?

**Comment by ke4roh** [comment]
I get the error when I try to add the datasource to a fresh install of Grafana.  I can query Graphite via curl without difficulty.

**Comment by torkelo** [comment]
Please more information! What settings did you use (direct or proxy) , what is the graphite request and response? Check chrome dev tools network tab can console,  supply full details etc.

**Comment by ke4roh** [comment]
![screenshot from 2016-04-20 10-31-55](https://cloud.githubusercontent.com/assets/713100/14678407/8c743ad4-06e3-11e6-8589-5eae2be92a3e.png)

![screenshot from 2016-04-20 11-00-12](https://cloud.githubusercontent.com/assets/713100/14679295/332682b2-06e7-11e6-9cae-529d3bec9b3e.png)

When I curl from my workstation, this gives me JSON of my data as I would expect:

    curl -u graphite:my_secret 'http://graphite-host.example.com:8080/metrics/find?query=*'

If I curl '...:8080/?query=*', which is what Chrome was attempting, it returns status 404 not found.

**Comment by torkelo** [comment]
Since you are using direct access the request happens from the browser directly to the datasource , the browser will check for CORS and that check fails , if you want to use direct your graphite server needside to allow CORS. Or you can use proxy

**Comment by noamcohen** [comment]
Another possible reason:
I cloned an Amazon EC2 machine which ran graphite/grafana perfectly, in a docker container.
When I ran the new instance, I saw the "Cannot read property 'message' of null" anywhere related to templates. 
The reason was found to be the IP of the data source: obviously it was not updated. Once I updated it ( to the IP of the machine running the rest of the graphite services), all went well.

the AWS machine is ubuntu 14.04
graphite version: I downloaded docker on 2016-11-03 which should contain the latest components (sorry - no more info)
